test/Casting/Casts.swift

```diff
@@ -1081,4 +1081,33 @@ CastsTests.test("type(of:) should look through __SwiftValue")
   expectEqual(t, "S")  // Fails: currently says `__SwiftValue`
 }
 
+#if _runtime(_ObjC)
+@objc protocol P106973771 {
+  func sayHello()
+}
+@objc protocol P106973771a { }
+extension NSObject : P106973771a {}
+CastsTests.test("Class metatype values should not cast to Obj-C existentials") {
+  @objc class C106973771: NSObject, P106973771 {
+    func sayHello() { print("Hello") }
+  }
+  // A class instance clearly conforms to this protocol
+  expectTrue(C106973771() is any P106973771)
+  // But the metatype definitely does not
+  expectFalse(C106973771.self is any P106973771)
+  // The cast should not succeed
+  expectNil(C106973771.self as? any P106973771)
+  // The following will crash if the cast succeeds
+  (C106973771.self as? any P106973771)?.sayHello()
+
+  // We should also follow ObjC rules for
+  // ObjC metatypes:
+  expectTrue(C106973771.self is any NSObjectProtocol)
+  expectTrue(C106973771.self is NSObject.Type)
+  expectTrue(C106973771.self is NSObject) // FIXME
```

Comment:
This line consistently fails.  I need to track down why; I suspect it might be that the compiler is optimizing this away differently than the runtime.